packages/core/src/service/account-management.ts

```diff
@@ -0,0 +1,460 @@
+    }
+    return await fcl.account(address);
+  }
+
+  async uploadMnemonicToGoogleDrive(
+    mnemonic: string,
+    username: string,
+    password: string
+  ): Promise<void> {
+    const isValidMnemonic = bip39.validateMnemonic(mnemonic);
+    if (!isValidMnemonic) {
+      throw new Error('Invalid mnemonic');
+    }
+    const app = getApp(process.env.NODE_ENV!);
+    const user = await getAuth(app).currentUser;
+    try {
+      // This would need to be imported from googleDriveService
+      await googleDriveService.uploadMnemonicToGoogleDrive(mnemonic, username, user!.uid, password);
+      mixpanelTrack.track('multi_backup_created', {
+        address: (await userWalletService.getCurrentAddress()) || '',
+        providers: ['GoogleDrive'],
+      });
+    } catch {
+      mixpanelTrack.track('multi_backup_creation_failed', {
+        address: (await userWalletService.getCurrentAddress()) || '',
+        providers: ['GoogleDrive'],
+      });
+    }
+  }
+
+  async verifyPasswordIfBooted(password: string): Promise<void> {
+    if (await keyringService.isBooted()) {
+      await keyringService.verifyPassword(password);
+    }
+  }
+
+  async createKeyringWithMnemonics(
+    publicKey: string,
+    signAlgo: number,
+    password: string,
+    mnemonic: string,
+    derivationPath = FLOW_BIP44_PATH,
+    passphrase = ''
+  ): Promise<any> {
+    await keyringService.clearCurrentKeyring();
+    const keyring = await keyringService.createKeyringWithMnemonics(
+      publicKey,
+      signAlgo,
+      password,
+      mnemonic,
+      derivationPath,
+      passphrase
+    );
+    return this._setCurrentAccountFromKeyring(keyring);
+  }
+
+  importPrivateKey = async (publicKey: string, signAlgo: number, password: string, pk: string) => {
+    const privateKey = ethUtil.stripHexPrefix(pk);
+    const buffer = Buffer.from(privateKey, 'hex');
+
+    const error = new Error('the private key is invalid');
+    try {
+      if (!ethUtil.isValidPrivate(buffer)) {
+        throw error;
+      }
+    } catch {
+      throw error;
+    }
+
+    const keyring = await keyringService.importPrivateKey(
+      publicKey,
+      signAlgo,
+      password,
+      privateKey
+    );
+    return this._setCurrentAccountFromKeyring(keyring);
+  };
+
+  async _setCurrentAccountFromKeyring(keyring: any, index = 0) {
+    const accounts = keyring.getAccountsWithBrand
+      ? await keyring.getAccountsWithBrand()
+      : await keyring.getAccounts();
+    const account = accounts[index < 0 ? index + accounts.length : index];
+
+    if (!account) {
+      throw new Error('the current account is empty');
+    }
+
+    const _account = {
+      address: typeof account === 'string' ? account : account.address,
+      type: keyring.type,
+      brandName: typeof account === 'string' ? keyring.type : account.brandName,
+    };
+    preferenceService.setCurrentAccount(_account);
+
+    return [_account];
+  }
+}
+
+export default new accountManagement();
```

Comment:
I reckon we should change all the exports to something like:

```suggestion
export const accountManagement = new AccountManagement();
```

Then we could export other things if need to more easily.

Reply:
Actually don't worry about this for now. Leave as you've done it. just change the Captilization of the Class name